src/query/ts/m3db/options.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/dbnode/encoding/m3tsz"
+	"github.com/m3db/m3/src/query/models"
+	"github.com/m3db/m3/src/query/pools"
+	"github.com/m3db/m3/src/query/ts/m3db/consolidators"
+)
+
+var (
+	defaultLookbackDuration = time.Duration(0)
+	defaultConsolidationFn  = consolidators.TakeLast
+	defaultIterAlloc        = func(r io.Reader) encoding.ReaderIterator {
+		return m3tsz.NewReaderIterator(r, m3tsz.DefaultIntOptimizationEnabled, encoding.NewOptions())
+	}
+)
+
+type encodedBlockOptions struct {
+	lookbackDuration time.Duration
+	consolidationFn  consolidators.ConsolidationFunc
+	tagOptions       models.TagOptions
+	bounds           *models.Bounds
+	iterAlloc        encoding.ReaderIteratorAllocate
+	pools            encoding.IteratorPools
+}
+
+// NewOptions creates a default encoded block options which dictates how
+// encoded blocks are generated.
+func NewOptions() Options {
+	return &encodedBlockOptions{
+		lookbackDuration: defaultLookbackDuration,
+		consolidationFn:  defaultConsolidationFn,
+		tagOptions:       models.NewTagOptions(),
+		iterAlloc:        defaultIterAlloc,
+		pools:            pools.BuildIteratorPools(),
+	}
+}
+
+func (o *encodedBlockOptions) SetLookbackDuration(
+	lookback time.Duration,
+) Options {
+	opts := *o
+	opts.lookbackDuration = lookback
+	return &opts
+}
+
+func (o *encodedBlockOptions) GetLookbackDuration() time.Duration {
+	return o.lookbackDuration
+}
+
+func (o *encodedBlockOptions) SetConsolidationFunc(
+	fn consolidators.ConsolidationFunc,
```

Comment:
Can you put these on one line? We should come up with a convention..

Reply:
And the other places where you only have one argument.

Reply:
Was going by line length mostly, put it  on a single line now

Should get a standard convention at some point though